src/app/widgets/Visualizer/Visualizer.jsx

```diff
@@ -1401,6 +1447,91 @@ class Visualizer extends Component {
         } else {
             setVisualizerReady();
         }
+
+        this.fileLoaded = true;
+    }
+
+    calculateLimits(data) {
+        /* get machine 0
+                0 is top right
+                1 is top left
+                2 bottom right
+                3 bottom left
+        */
+        let machineCorner = reduxStore.getState().controller.settings.settings.$23;
+
+        let xMultiplier = 1;
+        let yMultiplier = 1;
+        switch (machineCorner) {
+        case 0:
+            xMultiplier = -1;
+            yMultiplier = -1;
+            break;
+        case 1:
+            xMultiplier = 1;
+            yMultiplier = -1;
+            break;
+        case 2:
+            xMultiplier = -1;
+            yMultiplier = 1;
+            break;
+        case 3:
+        default:
+            xMultiplier = 1;
+            yMultiplier = 1;
+            break;
+        }
+
+        // get soft limits
+        let xMax = parseFloat(reduxStore.getState().controller.settings.settings.$130);
+        let yMax = parseFloat(reduxStore.getState().controller.settings.settings.$131);
+        let zMax = parseFloat(reduxStore.getState().controller.settings.settings.$132);
+
+        // get wpos
+        let wpos;
+        if (data !== 0) {
+            wpos = reduxStore.getState().controller.wpos;
```

Comment:
machinePosition and workPosition are already mapped as props to this component, no need to fetch them a second time